### DISCUSSION TITLE
Combining .ProjectInto<>() with Select() on >5.4.2?

### DISCUSSION THREAD

**Discussion by wagich** [ravendb/ravendb#15634] [Q&A] [answered]
After upgrading from 5.4.1 to the latest version, a query that was using `ProjectInto` (to retrieve stored fields) then `Select` on top of that stopped working and weirdly started throwing JsonSerializationException.

https://github.com/ravendb/ravendb/issues/15105 seems to be about the same thing, but was closed as "don't do that". My problem is that I have a reusable "base query" that by default has to retrieve fields stored only in the index (which doesn't work without ProjectInto). This query is then used from other places and might also be customized further using `Select()` before being executed.

Is it possible to restore the previous behaviour? IMHO it was more logical, especially because there isn't any indication that they can't be combined.
Or does anyone have a workaround that enables both retrieving stored index fields without specifying a projection and also enables customization?

For more context: I've isolated the behaviour change to commit https://github.com/ravendb/ravendb/commit/7e62f91f533dfe5ae36a20bec343504fb3e3876e
and created a test case at https://gist.github.com/wagich/8a1ad17e8b266d2176acf96636a50267

**Comment by garayx** [comment]
Hi
Both `ProjectInto()` & `Select()` are doing projection so you should have only one of them in the query.  

You are using `Select()` for customization, you can also return the stored fields by using `Select()`:
https://ravendb.net/docs/article-page/5.1/Csharp/indexes/querying/projections#example-ii---projecting-stored-fields

**Reply by ayende** [reply]
The actual problem here is that `Select` is _additive_. So you are actually returning all the fields of the item.
Then the deserialization code sees that there are multiple values there, and we don't try to get just the value to desarialize.

**Comment by ayende** [comment, marked as answer]
https://github.com/ravendb/ravendb/pull/15638

**Reply by wagich** [reply]
Thank you for the fast fix!
